src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,49 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Class to create a deadline for the user.
+ */
+public class Deadline extends Task {
+    private LocalDate date;
+    private boolean hasDate = false;
+
+    /**
+     * Constructor for Deadline.
+     *
+     * @param name label for the deadline.
+     * @param date Date inputted by the /by command.
+     */
+    public Deadline(String name, String date) {
+        super(name);
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MM-dd-yyyy");
+            this.date = LocalDate.parse(date.strip());
+            this.hasDate = true;
+        } catch (DateTimeParseException e) {
+            super.editName(String.format(" (by: %s)", date));
+        }
+    }
+
+    /**
+     * Returns string representation of deadline, including marked or not, date of
+     * deadline and contents.
+     *
+     * @return String representation of the deadline.
+     */
+    @Override
+    public String toString() {
+        String checkBox = super.isMarked() ? "[X]" : "[ ]";
+        char taskType1 = 'D';
```

Comment:
Would it be better if the variable name was `taskTypeDeadline` instead of `taskType1`, making it easier to differentiate from the String variable below of a similar variable name?
```suggestion
        char taskTypeDeadline = 'D';
```